ddclient.in

```diff
@@ -2146,6 +2154,128 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## geturl -- via curl
+######################################################################
+sub curl_cmd {
+    my $params = join("\n",@_);
+    my $tmpfile;
+
+    return '' if (length($params) == 0); ## no parameters provided
+
+    # debug("CURL:\r\n%s\r\n",$params); ## Uncomment only if required as may expose credentials
+
+    eval "no warnings 'all'; use File::Spec;";
+    $tmpfile = ((!$@) ? File::Spec->tmpdir() : '/tmp').'/ddclient-';
+    my @chars = ("A".."Z", "a".."z", "0".."9");
+    $tmpfile .= $chars[rand @chars] for 1..10;
+    debug("CURL Tempfile: %s",$tmpfile);
+    open(my $TFH, '>', $tmpfile) or fatal("Falied to open $tmpfile for curl");
+    chmod(0600,$tmpfile);
+    print($TFH $params);
+    close($TFH);
+    my $reply = qx{ curl --config $tmpfile 2>/dev/null };
```

Comment:
Check `$?` to see if curl completed successfully.

Reply:
Yes, agree.  Was just discussing that with @abelbeck we need to test for several fail conditions (like invalid certificate, malformed URL, etc.).  Will add that.

Reply:
Implemented this with a subset of the more likely return codes issuing a meaningful warning message.  If you think there are others that should be added to this shortlist let me know.